### PR TITLE
Update go to 1.17

### PR DESCRIPTION
**Pull request by muvaf**
### Description of your changes

Updates Go to 1.17.5

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested

N/A

### CODE REVIEW

File: cross/Dockerfile [hunk 1]
```diff
@@ -45,7 +45,7 @@ RUN curl -fsSL "https://awscli.amazonaws.com/awscli-exe-linux-x86_64-2.1.21.zip"
 
 # ------------------------------------------------------------------------------------------------
 # Go support
-RUN GO_VERSION=1.16.1 && \
+RUN GO_VERSION=1.17.5 && \
     GO_HASH=3edc22f8332231c3ba8be246f184b736b8d28f06ce24f08168d8ecf052549769 && \
```
Comment (hasheddan) [inline cross/Dockerfile:49]: @muvaf I believe this hash also needs to be updated

File: cross/Dockerfile [hunk 2]
```diff
@@ -45,7 +45,7 @@ RUN curl -fsSL "https://awscli.amazonaws.com/awscli-exe-linux-x86_64-2.1.21.zip"
 
 # ------------------------------------------------------------------------------------------------
 # Go support
-RUN GO_VERSION=1.16.1 && \
+RUN GO_VERSION=1.17.5 && \
```
Comment (hasheddan) [inline cross/Dockerfile:48]: Just curious, did you have a case where you were running into issues with Go version? I believe we have moved away from using cross container everywhere in Crossplane ecosystem

File: cross/Dockerfile (hunk 1 above)
Comment (muvaf) [reply cross/Dockerfile]: Thanks for catching!

**Review by hasheddan — APPROVED** [review]

File: cross/Dockerfile (hunk 2 above)
Comment (muvaf) [reply cross/Dockerfile]: No, I just wanted to upgrade starting from crossplane-runtime and found this line. Should we remove it altogether if it's not used in Upbound as well?

Comment (hasheddan) [reply cross/Dockerfile]: @muvaf I think we need to keep the cross container around for the time-being 👍🏻

**Pull request merged**